application/app/src/main/java/uk/nhs/digital/uec/api/authentication/cognito/CognitoIdpServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package uk.nhs.digital.uec.api.authentication.cognito;
+
+import static uk.nhs.digital.uec.api.authentication.localstub.LocalConstants.PASSWORD;
+import static uk.nhs.digital.uec.api.authentication.localstub.LocalConstants.USERNAME;
+import static uk.nhs.digital.uec.api.authentication.localstub.LocalConstants.USER_PASSWORD_AUTH;
+import static uk.nhs.digital.uec.api.util.Utils.calculateSecretHash;
+
+import com.amazonaws.services.cognitoidp.AWSCognitoIdentityProvider;
+import com.amazonaws.services.cognitoidp.model.AWSCognitoIdentityProviderException;
+import com.amazonaws.services.cognitoidp.model.InitiateAuthRequest;
+import com.amazonaws.services.cognitoidp.model.InitiateAuthResult;
+import com.amazonaws.services.cognitoidp.model.InvalidPasswordException;
+import com.amazonaws.services.cognitoidp.model.NotAuthorizedException;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.nhs.digital.uec.api.authentication.exception.InvalidCredentialsException;
+import uk.nhs.digital.uec.api.authentication.model.AuthToken;
+import uk.nhs.digital.uec.api.authentication.model.Credential;
+
+@Service
+@Slf4j
+public class CognitoIdpServiceImpl implements CognitoIdpService {
+
+  @Autowired private AWSCognitoIdentityProvider cognitoClient;
```

Comment:
could we not have private final as the access modifier

Reply:
This attribute is Autowired and so it can't be final